### ISSUE TITLE
Ship Hive Toronto privacy badges, pages and claim codes

### ISSUE THREAD

**Issue by smithisgeneric** [MozillaFoundation/plan#206]
Deliver badge designs, claim codes, and landing page(s) for three Hive Toronto badges.

## RACI

* Phase: Design & Dev
* Driver: @hannahkane
* Decision Maker: @smithisgeneric 
* Design Lead: @cassiemc 
* Development Lead: @ScottDowne 
* Quality Verifier: @iamjessklein  

## Deadline (UPDATED): Feb 26
* stretch goal: Feb 21 (educator / learner event)
* **actual target deadline: Feb 26 (funder event)**
* drop dead deadline: March 31(funder requirement) 

### Deliverables
- [x] **Design 3 badges** -- Privacy Coach, IP Tracer and Data Trail badge https://github.com/MozillaFoundation/plan/issues/302
- [x]  **Design 3 badge pages (optimized for claim codes)** https://github.com/MozillaFoundation/Mozilla-Learning/issues/10
- [x]  **Build those 3 badge pages. And ensure claim codes work.** https://github.com/MozillaFoundation/Mozilla-Learning/issues/11
- [ ]  **Create a landing page on hivetoronto.org.** Created using WordPress and will be linked out from http://hivetoronto.org/portfolio/privacybadges. Linking out to the three individual badge pages. https://github.com/MozillaFoundation/Mozilla-Learning/issues/12

**Comment by ldecoursy** [comment]
@smithisgeneric Is Feb. 21st a deadline per the grant?

**Comment by smithisgeneric** [comment]
Feb 21st is the knowledge mobilization workshop with educators where the curriculum is demoed (and hopefully the badges are issuable).  The hard deadline for all deliverables for the grant is March 31.

**Comment by hannahkane** [comment]
I recommend we consider this a P2 for the next heartbeat, and a P1 for the following if it doesn't get completed. 

I'm really not sure about the effort required (read: I don't know if this is "easy" or "hard") to reach the MVP, so perhaps a first step is for a couple people to get together to determine the specific work involved. Probably @smithisgeneric  + a developer who's somewhat familiar with badging (and I'd like to be there, too).

**Comment by smithisgeneric** [comment]
@hannahkane do I need to do anything else to signal a dev is needed here or are you on it?

**Comment by smithisgeneric** [comment]
@cassiemc can you provide some svg files of an existing badges (a couple of examples should do) currently on Webmaker to support this?

**Comment by cassiemc** [comment]
I'm confused @smithisgeneric. It doesn't seem like the scope of work is determined yet, so not sure what assets are needed. @iamjessklein has many of the assets too. (we should get those shared centrally)

You can see a full list of badges here if you're looking for something specific: https://blog.webmaker.org/badges

**Comment by smithisgeneric** [comment]
I was looking to collect an SVG (editable) art file for a badge or 2 (i.e., it could be the privacy badge and remix badge) so that I could have a contract team member (hiring in process) do the graphic design of the badges. 

The contract staff member is doing a limited number of teaching kit related hours, specifically tied to the [Hive Toronto privacy badges project] (http://hivetoronto.org/portfolio/privacybadges/)  

This can wait @cassiemc if I've jumped ahead.

**Comment by iamjessklein** [comment]
@smithisgeneric I believe [this link](https://www.dropbox.com/sh/0kdt7yh972fe760/AADoZvckqbzHOUjlHO2oNXA4a?dl=0) might be useful.

**Comment by cassiemc** [comment]
Assigning @iamjessklein as a P2 this heartbeat. Have updated the first comment to reflect this.

**Comment by cassiemc** [comment]
@hannahkane @ldecoursy This initiative needs an owner in the RACI at the top, should that be one of you two? 

(cc'ing @OpenMatt b/c of the US holiday...)

**Comment by iamjessklein** [comment]
If these badges are getting shipped on Webmaker.org - we will most likely need dev support at some point.

**Comment by hannahkane** [comment]
Discussed with @ldecoursy —I'll take this one.

@smithisgeneric and @iamjessklein—I'll send an invite for a meeting for the three of us.

**Comment by OpenMatt** [comment]
updated the RACI above -- please edit if wrong.

**Comment by hannahkane** [comment]
Meeting scheduled for 1/27 (late in the heartbeat, since this is a p2)

**Comment by iamjessklein** [comment]
wondering if these badges should match the webmaker badges/ the hive badges branding ... or be something totally different.

cc @cassiemc

**Comment by cassiemc** [comment]
@iamjessklein I'm sure a conversation would help but on first blush the title of the ticket – "Hive Toronto badges" – makes me think let's not introduce any new styles. Sorry if I'm missing some nuance here?

**Comment by OpenMatt** [comment]
+1 let's use existing styles / templates

**Comment by iamjessklein** [comment]
Yes, well if you recall we have two templates - the Hive one and the Webmaker one. I assume we should match these badges to the Hive ones.

**Comment by smithisgeneric** [comment]
A contract team members has submitted these badge visual ideas

![webmaker-training-badges-coach](https://cloud.githubusercontent.com/assets/6931814/5926452/1d332cc2-a639-11e4-8e10-b8ac728ebd80.png)
![webmaker-training-badges-datatrail](https://cloud.githubusercontent.com/assets/6931814/5926451/1d3188fe-a639-11e4-8628-ebc83a32d475.png)
![webmaker-training-badges-iptracer](https://cloud.githubusercontent.com/assets/6931814/5926453/1d34633a-a639-11e4-92d0-009900eb31c1.png)

**Comment by hannahkane** [comment]
@smithisgeneric, @iamjessklein, and I met today to discuss next steps which are:
 * Ashley will iterate on the badge designs, making use of the design standards established with the Hive badges, and making the icons Font Awesome-like. @iamjessklein and @cassiemc will provide feedback on and approve the final design.
 * We'll likely make use of the existing UX for applying for and issuing badges, and specifically the flow for using a claim code to earn a badge. @amirad and I are meeting tomorrow to write some tickets for making some small improvements to the UX. 
 * We'll likely need a landing page for these privacy badges, since they are connected to a specific initiative. If we decide that's needed, I'll create another ticket for it, so it can be assigned.

/cc @davidascher

**Comment by davidascher** [comment]
Found this which could be handy:

https://docs.google.com/spreadsheet/ccc?key=0AnUOR-0CBMsTdE1hWnJnYXlNdnktVUlhWWdjVzFVR0E&usp=gmail#gid=0

**Comment by davidascher** [comment]
Also see:
http://emilygoligoski.com/2014/05/07/introducing-mozilla-visual-templates-for-badge-issuers/

**Comment by smithisgeneric** [comment]
Ashley needs the EPS file for the Hive badge, consistent with https://webmaker.org/en-US/badges/hive-community-member cc: @iamjessklein @hannahkane @cassiemc

**Comment by hannahkane** [comment]
@iamjessklein @cassiemc  - can either of you provide the eps file? I don't have access to it.

**Comment by hannahkane** [comment]
Moving this to Feb 13 milestone, since it didn't get completed.

**Comment by OpenMatt** [comment]
@hannahkane : do we need a dev this Heartbeat? 
* can you also update https://wiki.mozilla.org/Webmaker/Badges with the relevant links David and others mention above, if useful?
* It already includes links like these, which should be helpful:
** Mozilla-wide badge style guide https://wiki.mozilla.org/File:Mozilla_Badges_project.pdf
** Web Literacy badge designs https://wiki.mozilla.org/Webmaker/Badges/Webliteracy

**Comment by OpenMatt** [comment]
* I deleted some cross-talk comments above to clean up this ticket
* @hannahkane : did we decided on a visual template to follow?

**Comment by cassiemc** [comment]
@hannahkane I don't have the Hive badge templates. @iamjessklein if you have them, can you post the link here and add them to our Design Resources folder on Drive? thanks!

**Comment by hannahkane** [comment]
I believe we decided on using the Hive Community Member badge as a template.

Updated the Badges page on the wiki. 

Will add "needs-dev" to this ticket.

**Comment by OpenMatt** [comment]
@cassiemc please add any relevant design drive links for badges here: https://wiki.mozilla.org/Webmaker/Badges

**Comment by cassiemc** [comment]
@iamjessklein ^^

**Comment by OpenMatt** [comment]
This ticket has grown long and a bit confusing. @hannahkane : some (totally optional) suggestions:
* **Groom / prune / edit this ticket.** Delete or edit old stuff that doesn't need to be here. Make it easier to grok. (imo: Don't be bashful about consolidating / pruning / deleting)
* **Schedule another kick-off meeting.** I know we had one last Heartbeat. But could be useful to synch up again. esp: hand-off from design to dev. (I could be wrong about this -- wasn't in the first one.) 
* **Clearly list out the deliverables.** e.g., (3 badge designs, + claim code functionality, + landing page, etc.) 
* **Update our badges documentation.** Which I know you've been doing already. It shouldn't be this difficult for a designer to find the style guide, templates, access design assets, etc.

**Comment by smithisgeneric** [comment]
Ashley needs the Hive Webmaker badge template EPS and Redpen Access to be able to proceed.  As to not list her personal email in the thread, I've emailed @cassiemc re: RedPen and @iamjessklein re: the EPS.

**Comment by iamjessklein** [comment]
[Here is the correct EPS file.](https://www.dropbox.com/s/rlzo0s4ywdh31cg/hive-badges.eps?dl=0) I am also going to add to the Google Drive. ﻿﻿

**Comment by hannahkane** [comment]
OK, I've cleaned up this ticket, updated the requirements in the description, and moved all documentation (including eps files of badge templates) to https://wiki.mozilla.org/Webmaker/Badges.

Next steps:

- [ ] @smithisgeneric to report on Ashley's progress with the badge design
- [ ] @cassiemc to assign a designer (assuming it will be @iamjessklein, but you know what they say about assuming) and @simonwex to assign an engineer to this project. The design and dev needs are to build a landing page that shows the three badges, and allows learners to enter a claim code for each. If the claim code is correct, the learner is automatically issued a badge, sees a success message, and the badge appears in their Webmaker profile. 
- [ ] @smithisgeneric to provide necessary copy for the landing page (including any language about the funder or partners)
- [ ] Have a meeting with the assigned designer and engineer to clarify the requirements. I'll schedule this once the roles are assigned.

**Comment by cassiemc** [comment]
Thanks for clarifying the scope of work @hannahkane! Somehow the 'landing page' request escaped me, I thought this was just designing the badges themselves. 

So, I can confirm @iamjessklein as designer for the page, and I can be quality for this – perhaps we need the actual badge designs as a subticket, to assign ashley there.

**Comment by hannahkane** [comment]
Thanks, @cassiemc . Added #302 to manage the badge design.

**Comment by smithisgeneric** [comment]
I made a v1 page of copy that could be the landing page for the badges https://docs.google.com/document/d/1bPJl4ch63JOQwmq2K3CjUdjkfzo3ajZKbzIYNqEdFsI/edit?usp=sharing (based on a previous example I found).  I can't find a good example of a "claims" page to write up copy. 

I think we need a dev assigned here as part of the process so that the functionality of the pages can be determined along with content (i.e., text, images, badges metadata).    cc: @iamjessklein @hannahkane

As a reminder/update: 
- Hive Toronto has a event where we want to use these badges on Feb. 21st 
- The funder has invited me to do a public presentation on this project in Ottawa Feb. 26th

**Comment by hannahkane** [comment]
+1 to needing a dev. @simonwex, can you please assign someone to this ticket?

Changed this to a p1 to reflect the external deadline.

**Comment by simonwex** [comment]
@hannahkane could you schedule a kickoff for this for tomorrow? @ScottDowne is going to help out.

**Comment by iamjessklein** [comment]
If this is shifting to a P1 and is being delivered on February 13 (in 3 days...) I am not confident that I can deliver on it as I am working on another P1 in this heartbeat.

**Comment by OpenMatt** [comment]
@iamjessklein I can help out as interim driver while @hannahkane is at NY work week. Working on getting up to speed now.

**Comment by OpenMatt** [comment]
* Added a separate sub-ticket for the Landing Page work: 
https://github.com/MozillaFoundation/Mozilla-Learning/issues/10

**Comment by OpenMatt** [comment]
* Having reviewed, I don't think it makes sense to label this a P1 right now, given the other design and dev priorities right now. It's a current P2, and should become a P1 once the next Heartbeat starts Monday. I'm going to update to that effect -- somebody speak up if they think that's wrong.

**Comment by smithisgeneric** [comment]
Downgrading to a P2 makes sense given today's discussion @OpenMatt

**Comment by OpenMatt** [comment]
* Thanks @smithisgeneric. I wasn't able to get everyone together for a proper kick-off on this today -- but I did have one-on-ones w. Karen, Scott & Jess.

**Comment by OpenMatt** [comment]
@hannahkane : I assume you're the driver for this one again, now that NY work week is over?

**Comment by hannahkane** [comment]
Yup, I can pick this one back up. Thanks for caretaking. :)

**Comment by cassiemc** [comment]
I'll just take this one and @iamjessklein can do design review (so jess can focus on MWC animations) :)

**Comment by iamjessklein** [comment]
thanks @cassiemc  :+1:

**Comment by cassiemc** [comment]
Design on the 3 badges is done, issue #302. Updated the checkbox in the description.

**Comment by iamjessklein** [comment]
R+ :+1: :shipit:

**Comment by ScottDowne** [comment]
I'm close to being able to make badges on staging.

I see the images are complete.

We need to get the images hosted somewhere, probably S3.

I also need to know other bits of meta data for each new badge, which would be nice in all one place.

Name:
Image URL:
Issuer URL:
Short description:
Long description:
Tags:
Criteria:

Criteria should be a list of steps. Can have multiple lists. The list can have a title. Example:

```
# This is an example title
* point 1
* point 2
* point 3
```

**Comment by hannahkane** [comment]
Thanks, @ScottDowne. How do we get images onto S3? I don't know the process for that.

@smithisgeneric, can you provide the requested metadata?

**Comment by ScottDowne** [comment]
@hannahkane I don't know off hand. I'll either need to bug jbuck or someone else to help me, or we need to just get someone else that knows how to just do it for us.

**Comment by ScottDowne** [comment]
So uploading the images won't be an issue and I'm ready to do that now.

I am concerned about the size and want to make sure @smithisgeneric has had a look at the images and is fine with the size of them: https://cloud.githubusercontent.com/assets/1186326/6250139/edf7648c-b759-11e4-9094-31a4be841a81.png

Compared to another badge: https://stuff.webmaker.org/badges/webmaker-mentor.png

I kinda feel like they should be the same, no?

Another concern about claim codes.

There are two types. A single use claim code attached to an email.

And a multi use claim code that anyone can claim once they get the code.

For a hectic event, the multi use claim code seems ideal but it does mean people can share the claim code to anyone else.

A single use claim code wouldn't have that issue but would require an admin creating a claim code and giving them out.

**Comment by cassiemc** [comment]
I can make them the same size now that I have the dimensions.

**Comment by cassiemc** [comment]
@ScottDowne do you think we should be using svg for these? would make them crisp at any size!

**Comment by ScottDowne** [comment]
@cassiemc works for me!

**Comment by cassiemc** [comment]
@ScottDowne ok great. SVG assets are here: https://www.dropbox.com/sh/epep6cvv2soi5dv/AABnM5MeRC6G51blZjBxx1PKa?dl=0

**Comment by hannahkane** [comment]
In an earlier meeting @smithisgeneric ok'd the multi-use codes, so I think we're good with that option..

**Comment by ScottDowne** [comment]
So it looks like we have a third option with claim codes. We can ahead of time create a bunch of single use claim code that we can give out. I think that would be better than the single use claim code.

**Comment by hannahkane** [comment]
@smithisgeneric - you're the decision-maker on this, so I'll leave this up to you.

Needed: 
 - [ ] decision about whether to use single-use, single-use w/ pre-generated codes, or multi-use claim codes.
 - [ ] Meta data detailed in comment above

**Comment by smithisgeneric** [comment]
* Multi-use codes meet the need 

I will get to the meta data asap (Hive Toronto has an event on, starting shortly)

**Comment by smithisgeneric** [comment]
@ScottDowne @hannahkane 

I have listed metadata here and noted questions via the comments:
https://docs.google.com/document/d/1SYumJMeGZ4oX8bDGVH4TkPbj9buYxJasYO-kwW5UZ8E/edit# 

This would benefit from review from someone familiar with the typical look + feel of metadata

**Comment by ScottDowne** [comment]
https://github.com/mozilla/webmaker.org/pull/1280 is the patch for the changes to the claim code

**Comment by ScottDowne** [comment]
We have badges on staging.

https://webmaker.mofostaging.net/en-US/badges/71b6498a0f7ebbdd401621234ff91dd1
https://webmaker.mofostaging.net/en-US/badges/6b8859a189d9de0a91ef2b163a7cbbcf
https://webmaker.mofostaging.net/en-US/badges/ab953d54e18ee1b49c4cc54de8224a29

Images are not yet uploaded, I'm just linking directly to dropbox... Seems to work well enough for staging.

The claim code related changes are in review.

The url slugs need to be updated manually in the database. I can do that last.

**Comment by ScottDowne** [comment]
- [ ] Upload images to s3
- [ ] Create badges on production
- [ ] Update production badge url slugs to be words and not an id
- [ ] Create a claim code on production badges

**Comment by smithisgeneric** [comment]
@ScottDowne I've updated the metadata again with the text to

"Get started with the [X] activity kit" in the long description as per your advice.  The funder's thank you was also put in the long description field.  Updates at https://docs.google.com/document/d/1SYumJMeGZ4oX8bDGVH4TkPbj9buYxJasYO-kwW5UZ8E/edit#

**Comment by ScottDowne** [comment]
https://stuff.webmaker.org/badges/hive-toronto-privacy-badges-coach.svg
https://stuff.webmaker.org/badges/hive-toronto-privacy-badges-data-trail.svg
https://stuff.webmaker.org/badges/hive-toronto-privacy-badges-ip-tracer.svg

Badges uploaded.

**Comment by ScottDowne** [comment]
Putting links in badge descriptions is a bit of an issue.

Adding a new field to the badge is also an issue.

I'm trying to find a good solution.

**Comment by ScottDowne** [comment]
Nvm, all good, markdown works in descriptions.

**Comment by hannahkane** [comment]
Looks like good progress today! thanks, everyone.

A few questions for @ScottDowne :
 - once the claim code-related changes are pushed, will the "What qualifies you to earn this badge?" question go away?
 - should I be seeing the criteria on the badge pages? I don't yet.
 - what's the new field you referenced?

**Comment by ScottDowne** [comment]
No new field, that was confusion.

There was a request to add a link, and our options included a new field but ended up not needing to go there. It would have been problematic if that was the only option.

**Comment by smithisgeneric** [comment]
Just flagging for @hannahkane @ScottDowne that I will have poor availability on Wednesday, Thursday and part of Friday this week (Feb. 25-27th) to respond to GitHub issues and new comments here.  I am traveling and attending a conference to present on the privacy badges project.

**Comment by ScottDowne** [comment]
All this is blocked on is making a few changes to the prod badges and landing one patch.

Both of these need help from @k88hudson 

For now I've moved onto p2s.

**Comment by k88hudson** [comment]
@thecount is there a list of bugs related to this?

**Comment by k88hudson** [comment]
@ScottDowne all of these maybe? https://github.com/mozilla/webmaker.org/issues?q=is%3Aopen+is%3Aissue+label%3Abadges

**Comment by ScottDowne** [comment]
Just this: https://github.com/mozilla/webmaker.org/pull/1280

Doing the data base changes was never filed.

**Comment by ScottDowne** [comment]
That's a list, but not one that is blocking this issue.

**Comment by k88hudson** [comment]
@ScottDowne :+1:

**Comment by hannahkane** [comment]
Hey @ScottDowne and @k88hudson — any movement on this today? Do I need to file a ticket for database changes?

**Comment by ScottDowne** [comment]
@hannahkane yes, the prod badges exist but don't have a pretty url and are hidden under a not so pretty url.

As of last night I have what I need to make the changes to the urls.

I am landing the patch to handle the claim code tomorrow.

Today I was trying something for a better way to update the urls. While it'll be useful going forward, I discovered already existing badges still need to have their urls updated the old way.

This ticket is fine for tracking changes.

**Comment by ScottDowne** [comment]
I think the summary is it should be all said and done tomorrow.

**Comment by k88hudson** [comment]
:+1: nice work @ScottDowne 
:cat2: :cat2: :cat2: :cat2: :cat2:

**Comment by ScottDowne** [comment]
https://webmaker.org/en-US/badges/privacy-coach
https://webmaker.org/en-US/badges/ip-address-tracer
https://webmaker.org/en-US/badges/data-trail-timeline

Ping me for the claim codes.

The claim code patch is blocked on some other stuff that's being sorted out now.

Claim codes still work if you go into the apply button in those pages.

**Comment by k88hudson** [comment]
:dancer: awesome!

**Comment by smithisgeneric** [comment]
What's the situation on listing criteria under Badge Requirements?  It seems the criteria should appear, or the heading Badge Requirements should be hidden @ScottDowne and cc: @hannahkane  

"1) Privacy literacy

You considered privacy as a component of web literacy or digital media literacy more broadly.  

2) Production-centered learning

You completed the learning activities in the teaching kit which involve designing, experimenting, exploring or other production-centered learning.

3) Discussion or reflection

You participated in discussion or independent reflection to consider privacy and your learning experience."

**Comment by ScottDowne** [comment]
Criteria is attached to the concept of applying.

Listing criteria and not letting people apply for the badge doesn't seem fair to me.

So I decided to hide it for badges like this.

I know we talked briefly about putting criteria in the learning kit instead.

**Comment by ScottDowne** [comment]
Err, to clarify that last comment a bit.

The hiding bit is still waiting on webmaker.org being built before it's on production.

I'm experimenting with hiding the apply button and keeping the criteria as a list, so we can do that if we want.

**Comment by hannahkane** [comment]
@ScottDowne - in the model you're experimenting with, would an authenticated user just see the three badge requirements + the "Have a claim code?" field + a Submit button? And, if that's the case, what would an unauth'ed user see?

**Comment by ScottDowne** [comment]
Yes, while logged in you see a "have a claim code" and the requirements, while not logged in, you see a "login to claim" instead, clicking it, prompts login.

**Comment by ScottDowne** [comment]
Should be all done.

https://webmaker.org/en-US/badges/privacy-coach
https://webmaker.org/en-US/badges/ip-address-tracer
https://webmaker.org/en-US/badges/data-trail-timeline

Me and @smithisgeneric have the claim codes.

The model I was experimenting with ended up being the one to use which is live now.

**Comment by hannahkane** [comment]
awsm. Thanks, @ScottDowne. Seems like all that's left is building the landing page (https://github.com/MozillaFoundation/Mozilla-Learning/issues/12). @smithisgeneric - do you need any add'l support on that one?

**Comment by smithisgeneric** [comment]
@ScottDowne @hannahkane I'm running through the Webmaker pages now (...waiting for Scott to resend the claims codes). The second bullet point for each criteria, for each of the 3 badges, doesn't make sense because there isn't a field/box to enter a response.  See an example in the screenshot.     

![criteriagithub](https://cloud.githubusercontent.com/assets/6931814/6398281/8d682786-bdb8-11e4-829d-c874e8c15bba.png)
 
I have limited ability to reply this afternoon due to my conference panel and lack of power outlets at my conference :)

**Comment by ScottDowne** [comment]
Done

**Comment by hannahkane** [comment]
Closing this ticket, as no more work is needed from design or engineering teams. The one outstanding ticket will be closed by @smithisgeneric early next week.
